README.md

```diff
@@ -0,0 +1,133 @@
+# Action Server
+
+The action server is an actionlib server for managing the execution of high level tasks and their semantic chaining.
+It takes actionlib goals describing series of high level tasks in json form, checks the consistency of the semantics, and executes them.
+This readme explains the architecture of the action server, the procedure of handling a goal and the lifecycle of an action.
+
+## Architecture
+
+![Action Server architecture](doc/action_server_architecture.jpg)
+
+The action server uses an actionlib interface for communication with its clients.
+A client implementation is provided as abstraction from the raw actionlib client.
+Current examples of clients to the action server are:
+ - The GPSR challenge
+ - The Natural Language Console
+
+Consult one of these implementations, or the Client docstring for example usage of the Client class.
+
+Aside from the actionlib SimpleActionServer, the Action Server holds an instance of the Task Manager.
+As the name suggests, this is the component that does the actual managing of the task (including its subtasks) for us.
+
```

Comment:
Above you explained how clients access the action server, at this point I would expect a global description of the action server itself. It gives you room to explain what things it has available (knowledge, smach state machines, available actions), and why these things are here (like "smach state machimes will do the actual work on behalf of the action").